### PR TITLE
[CWS] Add metric on ring buffer usage

### PR DESCRIPTION
**Pull request by lebauce**
### What does this PR do?

Add a metric datadog.runtime_security.perf_buffer.bytes.free that measures the free
space of an eBPF ring buffer.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/security/ebpf/c/defs.h [hunk 1]
```diff
@@ -369,6 +387,7 @@ void __attribute__((always_inline)) send_event_with_size_ptr(void *ctx, u64 even
     int perf_ret;
     if (use_ring_buffer) {
         perf_ret = bpf_ringbuf_output(&events, kernel_event, kernel_event_size, 0);
+        store_ring_buffer_stats();
```
Comment (Gui774ume) [inline pkg/security/ebpf/c/defs.h:390]: 👋🏻 Should we call this from an eRPC call maybe ?

File: pkg/security/metrics/metrics.go [hunk 2]
```diff
@@ -101,6 +101,9 @@ var (
 	// MetricPerfBufferBytesRead is the name of the metric used to count the number of bytes read from a perf buffer
 	// Tags: map
 	MetricPerfBufferBytesRead = newRuntimeMetric(".perf_buffer.bytes.read")
+	// MetricPerfBufferBytesFree is the name of the metric used to count the number of bytes left in the ring buffer
+	// Tags: map
+	MetricPerfBufferBytesFree = newRuntimeMetric(".perf_buffer.bytes.free")
```
Comment (Gui774ume) [inline pkg/security/metrics/metrics.go:106]: We discussed offline about:
- changing this metric to a usage percentage
- tagging this metric with the discrete size of the maps

File: pkg/security/metrics/metrics.go [hunk 3]
```diff
@@ -101,6 +101,9 @@ var (
 	// MetricPerfBufferBytesRead is the name of the metric used to count the number of bytes read from a perf buffer
 	// Tags: map
 	MetricPerfBufferBytesRead = newRuntimeMetric(".perf_buffer.bytes.read")
+	// MetricPerfBufferBytesUsed is the name of the metric used to count the percentage of space left in the ring buffer
+	// Tags: map
+	MetricPerfBufferBytesUsed = newRuntimeMetric(".perf_buffer.bytes.used")
```
Comment (Gui774ume) [inline pkg/security/metrics/metrics.go:106]: `Used` feels like if they can't be reused ... what about `in_use` ? Sounds mode like "Datadog" 😅

**Review by Gui774ume — APPROVED** [review]
LGTM

File: pkg/security/metrics/metrics.go (hunk 3 above)
Comment (paulcacheux) [reply pkg/security/metrics/metrics.go]: Fixed in latest commit

**Pull request merged**